csrc/device_lower/pass/circular_buffer.cpp

```diff
@@ -215,6 +217,711 @@ class CircularBufferLoopCloner : public kir::IrVisitor {
   const std::unordered_set<Expr*>& exclude_;
 };
 
+// Description:
+// Replicates circular buffer loops for Prologue, Main, and
+// Epilogue. Prologue only copies the load expressions of circular
+// buffered tensors, whereas Epilogue does any expression other than
+// the loads. Main copies everything.
```

Comment:
Mention pre-prologue and post-prologue as well, to show that we wind up with five loops.